packages/strapi-plugin-users-permissions/admin/src/containers/EditPage/actions.js

```diff
@@ -0,0 +1,103 @@
+/*
+ *
+ * EditPage actions
+ *
+ */
+import { fromJS, List, Map } from 'immutable';
+import { concat, get } from 'lodash';
+import {
+  ADD_USER,
+  GET_PERMISSIONS,
+  GET_PERMISSIONS_SUCCEEDED,
+  GET_ROLE,
+  GET_ROLE_SUCCEEDED,
+  ON_CANCEL,
+  ON_CHANGE_INPUT,
+  ON_CLICK_DELETE,
+  SET_FORM,
+} from './constants';
+
+export function addUser(newUser) {
+  return {
+    type: ADD_USER,
+    newUser,
+  };
+}
+
+export function getPermissions() {
+  return {
+    type: GET_PERMISSIONS,
+  };
+}
+
+export function getPermissionsSucceeded(data) {
+  const permissions = Map(fromJS(data.permissions));
+
+  return {
+    type: GET_PERMISSIONS_SUCCEEDED,
+    permissions,
+  };
+}
+
+export function getRole(id) {
+  return {
+    type: GET_ROLE,
+    id,
+  };
+}
+
+export function getRoleSucceeded(data) {
+  const form = Map({
+    name: get(data, ['role', 'name']),
+    description: get(data, ['role', 'description']),
+    users: List(get(data, ['role', 'users'])),
+    permissions: Map(fromJS(get(data, ['role', 'permissions']))),
+  });
+
+  return {
+    type: GET_ROLE_SUCCEEDED,
+    form,
+  };
+}
+
+export function onCancel() {
+  return {
+    type: ON_CANCEL,
+  };
+}
+
+export function onChangeInput({ target }) {
+  const keys = concat(['modifiedData'], target.name.split('.'));
```

Comment:
You don't really Lodash in this case native function... It's useful to use Lodash where the library fixes a weird JS behaviour or help a lot such as (_.difference, _.pick, etc).

```js
const keys = ['modifiedData'].concat(target.name.split('.'));
```